### DISCUSSION TITLE
Support of Policy's custom response

### DISCUSSION THREAD

**Discussion by mybouhssina** [filamentphp/filament#14910] [Ideas]
Currently, filament returns 403 no matter what is the response of the policy (like 404/denyAsNotFound)
```php
public static function can(string $action, ?Model $record = null): bool
    {
        if (static::shouldSkipAuthorization()) {
            return true;
        }

        $model = static::getModel();

        try {
            return authorize($action, $record ?? $model, static::shouldCheckPolicyExistence())->allowed();
        } catch (AuthorizationException $exception) {
            return $exception->toResponse()->allowed(); // due to this line
        }
    }
```
Is there a way to allow custom responses from the policy?

**Comment by tonypartridge** [comment]
Policies don't trigger the responses, since it's already found it exists. The route also exists... So since a policy is being hit, it will be a 403 or a 200....

**Reply by mybouhssina** [reply]
@tonypartridge I see what you mean, but in some cases you don't want the user to even know that the page exists, check this part of Laravel doc:
https://laravel.com/docs/11.x/authorization#customizing-policy-response-status
